testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -568,6 +569,136 @@ public void testTrinoWritingDataWithObjectStorageLocationProvider(StorageFormat
+                        trinoTableName,
+                        storageFormat));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES %s", trinoTableName, TRINO_INSERTED_PARTITION_VALUES));
+
+        assertThat(onSpark().executeQuery("SELECT * FROM " + sparkTableName)).containsOnly(EXPECTED_PARTITION_VALUES);
+        assertThat(onTrino().executeQuery("SELECT * FROM " + trinoTableName)).contains(EXPECTED_PARTITION_VALUES);
+        onTrino().executeQuery("DROP TABLE " + trinoTableName);
+    }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProvider = "storageFormats")
+    public void testStringPartitioningWithSpecialCharactersInsertInSpark(StorageFormat storageFormat)
+    {
+        String baseTableName = "test_string_partitioning_with_special_chars_ctas_in_spark";
+        String trinoTableName = trinoTableName(baseTableName);
+        String sparkTableName = sparkTableName(baseTableName);
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + trinoTableName);
+        onTrino().executeQuery(
+                format("CREATE TABLE %s (id BIGINT, part_col VARCHAR) WITH (partitioning = ARRAY['part_col'], format = '%s')",
+                        trinoTableName,
+                        storageFormat));
+        onSpark().executeQuery(format("INSERT INTO %s VALUES %s", sparkTableName, SPARK_INSERTED_PARTITION_VALUES));
+
+        assertThat(onSpark().executeQuery("SELECT * FROM " + sparkTableName)).containsOnly(EXPECTED_PARTITION_VALUES);
+        assertThat(onTrino().executeQuery("SELECT * FROM " + trinoTableName)).contains(EXPECTED_PARTITION_VALUES);
+        onTrino().executeQuery("DROP TABLE " + trinoTableName);
+    }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProvider = "storageFormats")
+    public void testStringPartitioningWithUtfCharsFromTrino(StorageFormat storageFormat)
+    {
+        String baseTableName = "test_string_partitioning_with_utf_chars_from_trino";
+        String trinoTableName = trinoTableName(baseTableName);
+        String sparkTableName = sparkTableName(baseTableName);
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + trinoTableName);
+        onTrino().executeQuery(format("CREATE TABLE %s (id BIGINT, part_col VARCHAR) WITH (partitioning = ARRAY['part_col'], format = '%s')", trinoTableName, storageFormat));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES (1, 'łąka')", trinoTableName));
+
+        assertThat(onTrino().executeQuery(format("SELECT part_col FROM %s WHERE id = 1", trinoTableName))).containsOnly(row("łąka"));
+        assertThat(onSpark().executeQuery(format("SELECT part_col FROM %s WHERE id = 1", sparkTableName))).containsOnly(row("łąka"));
+        onTrino().executeQuery("DROP TABLE " + trinoTableName);
+    }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProvider = "storageFormats")
+    public void testStringPartitioningWithUtfCharsFromSpark(StorageFormat storageFormat)
```

Comment:
can we do all this with extending TRINO_INSERTED_PARTITION_VALUES list, and thus with one method?